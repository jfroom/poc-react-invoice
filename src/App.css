.navbar.navbar-inverse {
  border-radius: 0;
  font-family: 'Roboto Slab', serif;
}
.navbar-header {
  float: left;
}
.navbar-right {
  float: right;
  margin-right: -15px;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
