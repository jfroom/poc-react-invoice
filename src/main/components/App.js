// @flow
import React from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { withRouter } from 'react-router'
import Header from './Header'
import {
  Breadcrumb,
  InvoiceFormContainer,
  InvoiceListContainer,
} from '../../invoices'
import styled from 'styled-components'

const StyleWrap = styled.div`
  font-family: 'Open Sans', Arial, sans-serif;
  h1, h2, h3, h4, h5, h6 {
    font-family: 'Roboto Slab', serif;
  }
`

const App = () => {
  const BreadcrumbWithRouter:Router = withRouter(Breadcrumb)

  // For GitHub Pages (prod), assign basename
  let basename = ''
  if (/https?:\/\/.+github\.io/.test(window.location.origin)) {
    basename = '/' + window.location.pathname.split('/')[1]
  }
  return (
    <Router basename={basename}>
      <StyleWrap>
        <Header />
        <div className="App container text-left">
          <BreadcrumbWithRouter />
          <Route exact path="/" component={InvoiceListContainer} />
          <Route path="/new" component={InvoiceFormContainer} />
          <Route path="/edit" component={InvoiceFormContainer} />
        </div>
      </StyleWrap>
    </Router>
  )
}
export default App
