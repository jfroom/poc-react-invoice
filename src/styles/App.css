/* Prefix with .App to give precedence over bootstrap styles */
.App h1, .App h2, .App h3, .App h4, .App h5, .App h6 {
  font-family: 'Roboto Slab', serif;
}
.navbar.navbar-inverse {
  border-radius: 0;
  font-family: 'Roboto Slab', serif;
}
.navbar-header {
  float: left;
}
.navbar-right {
  float: right;
  margin-right: -15px;
}
.invoice-list td:nth-child(2){
  width: 50%;
}
.line-items .form-group {
  margin-bottom: 0;
}
.line-items td:nth-child(1) {
  width: 75%;
}
.line-items td:nth-child(3){
  width: 15%;
}
.line-items td:nth-child(3){
  width: 10%;
  vertical-align: middle;
}
.btn {
  margin: 2px;
}
