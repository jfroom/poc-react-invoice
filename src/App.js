import React, { Component } from 'react';
import {Navbar} from 'react-bootstrap';
import './App.css';

// HEADER
function Header() {
  return (
    <Navbar bsStyle='inverse'>
      <div className="container">
        <div className="navbar-header">
          <a className="navbar-brand" href="#">Invoice POC</a>
        </div>
        <ul className="nav navbar-nav navbar-right">
          <li>
            <a href="https://github.com/jfroom/poc-react-invoice"><small>Source</small></a>
          </li>
        </ul>
      </div>
    </Navbar>
  );
}

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <p>content here</p>
      </div>
    );
  }
}

export default App;
