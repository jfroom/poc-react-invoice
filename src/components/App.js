import React, { Component } from 'react';
import {BrowserRouter as Router, Route} from 'react-router-dom'
import { withRouter } from 'react-router'

import Header from './Header'
import Breadcrumb from './Breadcrumb'
import InvoiceListContainer from '../containers/InvoiceListContainer'
import InvoiceFormContainer from '../containers/InvoiceFormContainer'
import '../styles/App.css';

class App extends Component {
  render() {
    const BreadcrumbWithRouter = withRouter(Breadcrumb)
    return (
      <Router>
        <div>
          <Header />
          <div className='App container text-left'>
              <BreadcrumbWithRouter/>
              <Route exact path="/" component={InvoiceListContainer} />
              <Route path="/new" component={InvoiceFormContainer} />
              <Route path="/edit" component={InvoiceFormContainer} />
          </div>
        </div>
      </Router>
    );
  }
}

export default App;
