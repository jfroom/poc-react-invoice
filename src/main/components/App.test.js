import React from 'react'
import App from './App'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const wrap = shallow(<App/>)
  expect(toJson(wrap)).toMatchSnapshot()
})
