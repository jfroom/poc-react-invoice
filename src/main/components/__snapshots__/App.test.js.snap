// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BrowserRouter
  basename=""
>
  <styled.div>
    <Header />
    <div
      className="App container text-left"
    >
      <withRouter(Breadcrumb) />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/new"
      />
      <Route
        component={[Function]}
        path="/edit"
      />
    </div>
  </styled.div>
</BrowserRouter>
`;
